cloud_functions/fastly_announce_purge/src/main.py

```diff
@@ -42,3 +53,97 @@ def purge_for_announce(cloud_event: CloudEvent):
         else:
             logger.warning(f"Announcement event caught, but no environment to purge cache. ENVIRONMENT: {environment}")
 
+
+def _purge_announced_papers():
+    """this function purges fastly's data for papers that have been created or updated since the last announcement"""
+    announcements= _get_days_announcements()
+    keys=_process_announcements(announcements)
+
+    #send purge request(s) to appropriate fastly services
+    environment = os.environ.get('ENVIRONMENT')
+    if environment == "PRODUCTION":
+        purge_fastly_keys(keys)
+        purge_fastly_keys(keys,"export.arxiv.org")
+    elif environment == "DEVELOPMENT":
+        purge_fastly_keys(keys, "browse.dev.arxiv.org")
+    elif environment == "TESTING":
+        logger.info(f"In TESTING enviroment. Would have purged keys {len(keys)}: {keys}\nAbove keys ({len(keys)}) not purged, in TESTING enviroment.")
+    return
+
+def _get_days_announcements()-> List[Tuple[str, int, str, str, str]]:
+    """gets data for most recent days announcement
+    return values are (paper_id, version, type of announcement, the papers category string, extra data (contains newly crosslisted categories))
+    """
+    mail= aliased(NextMail)
+    meta=aliased(Metadata)
+    today=session.query(func.max(mail.mail_id)).scalar_subquery()
+    result = (
+        session.query(mail.paper_id, mail.version, mail.type, meta.abs_categories, mail.extra)
+        .join(meta, mail.document_id == meta.document_id)
+        .filter(mail.mail_id == today)
+        .filter(meta.is_current==1)
+        .all()
+    )
+    return result
+
+def _process_announcements(announcements:List[Tuple[str, int, str, str, str]])->List[str]:
+    """ Processes the data for the mailing table to find the keys needed for each entry
+    returns list of keys to purge
+    parameters values are (paper_id, version, type of announcement, the papers category string, extra data (contains newly crosslisted categories))
+    returns a list of all keys to purge
+    """
+    keys=[]
+    lists=set() #also includes year pages, kept as set because of potential duplicates
```

Comment:
Hold on, don't throw shade at `set` it is a great data type. No justifications needed.